include/arch/posix/linker.ld

```diff
@@ -25,19 +25,28 @@ SECTIONS
 /* Located in generated directory. This file is populated by the
  * zephyr_linker_sources() Cmake function.
  */
-#include <snippets-rom-start.ld>
+SECTION_PROLOGUE(rom_start,,)
+{
+	#include <snippets-rom-start.ld>
+} GROUP_LINK_IN(ROMABLE_REGION)
 
 #include <linker/common-rom.ld>
 
 /* Located in generated directory. This file is populated by the
  * zephyr_linker_sources() Cmake function.
  */
-#include <snippets-rodata.ld>
+SECTION_PROLOGUE(rodata,,)
+{
+	#include <snippets-rodata.ld>
+} GROUP_LINK_IN(ROMABLE_REGION)
 
 /* Located in generated directory. This file is populated by the
  * zephyr_linker_sources() Cmake function.
  */
-#include <snippets-rwdata.ld>
+SECTION_DATA_PROLOGUE(rwdata,,)
+{
+	#include <snippets-rwdata.ld>
+} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
```

Comment:
GROUP_*_LINK_IN are nops for the posix arch, and having symbols pointing to one region in the elf but expected in another at runtime is a bit of a funny concept also in this case.
Are you sure this is what you want to do?

Reply:
simply because this is how those snippets are placed in other archs, meaning that if we later decides that those macros should behave differently for native_posix then we won't have to change this code.